test/barback/dartdevc/module_computer_test.dart

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:barback/barback.dart';
+import 'package:test/test.dart';
+
+import 'package:pub/src/barback/dartdevc/module_computer.dart';
+
+import 'util.dart';
+
+main() {
+  group('computeModules', () {
+    test('no strongly connected components, one shared lib', () async {
+      var assets = makeAssets({
+        'a|lib/a.dart': '''
+          import 'b.dart';
+          import 'src/c.dart';
+        ''',
+        'a|lib/b.dart': '''
+          import 'src/c.dart';
+        ''',
+        'a|lib/src/c.dart': '''
+          import 'd.dart';
+        ''',
+        'a|lib/src/d.dart': '''
+        ''',
+      });
+
+      var expectedModules = [
+        equalsModule(makeModule(
+            package: 'a',
+            name: 'lib__a',
+            srcs: ['a|lib/a.dart'],
+            directDependencies: ['a|lib/b.dart', 'a|lib/src/c.dart'])),
+        equalsModule(makeModule(
+            package: 'a',
+            name: 'lib__b',
+            srcs: ['a|lib/b.dart'],
+            directDependencies: ['a|lib/src/c.dart'])),
+        equalsModule(makeModule(
+            package: 'a',
+            name: 'lib__a\$lib__b',
```

Comment:
What does this name mean? What does the module containing c and d have to do with a and b?